mcs_benchmark_data/pipelines/commonsense_qa/commonsense_qa_benchmark_transformer.py

```diff
@@ -23,125 +24,118 @@
-        dev_jsonl_file_path: Path,
-        test_jsonl_file_path: Path,
-        train_jsonl_file_path: Path,
-        submission_data_jsonl_file_path: Path,
-        submission_jsonl_file_paths: Tuple[Path, ...]
+        self, benchmark_json_file_path: Path, **kwds
     ) -> Generator[_Model, None, None]:
 
-        benchmark_json = open(benchmark_json_file_path)
-        yield from self.__transform_benchmark(benchmark_json)
-
-        # Yield benchmark dev sample
-        dev_json = open(dev_jsonl_file_path)
-        yield from self.__transform_benchmark_sample(dev_json, "dev")
-
-        # Yield benchmark train sample
-        train_json = open(train_jsonl_file_path)
-        yield from self.__transform_benchmark_sample(train_json, "train")
-
-        # Yield benchmark test sample
-        test_json = open(test_jsonl_file_path)
-        yield from self.__transform_benchmark_sample(test_json, "test")
-
-
-    def __transform_benchmark(self, benchmark_json) -> Generator[_Model, None, None]:
-        # benchmark_bootstrap = BenchmarkBoostrap.from_json(benchmark_json)
-        benchmark_metadata = json.load(benchmark_json)
+        with open(benchmark_json_file_path) as benchmark_json:
+            benchmark_metadata = json.load(benchmark_json)
 
         benchmark_datasets = []
 
         for dataset in benchmark_metadata["datasets"]:
 
             dataset_type = dataset["@id"].split("/")[-1]
 
-            if dataset_type == "train":
-                new_dataset = BenchmarkTrainingDataset(
-                    uri=dataset["@id"], type = "train", name=dataset["name"], entries=tuple()
-                )
-            elif dataset_type == "test":
-                new_dataset = BenchmarkTestDataset(
-                    uri=dataset["@id"], type = "test", name=dataset["name"], entries=tuple()
-                )
-            elif dataset_type == "dev":
-                new_dataset = BenchmarkDevDataset(
-                    uri=dataset["@id"], type = "dev", name=dataset["name"], entries=tuple()
-                )
+            dataset_uri = "{}:dataset:{}".format(self.__URI_BASE, dataset["@id"])
 
-            yield new_dataset
+            new_dataset = self.__BENCHMARK_DATASET_CLASSES[dataset_type](
+                uri=dataset_uri, type=dataset_type, name=dataset["name"]
+            )
 
             benchmark_datasets.append(new_dataset)
 
-        yield Benchmark(
-            uri="",
+            yield from self.__transform_benchmark_sample(
+                kwds[dataset_type + "_jsonl_file_path"], dataset_type, new_dataset.uri
+            )
+
+        benchmark = Benchmark(
+            uri="{}:benchmark:{}".format(self.__URI_BASE, benchmark_metadata["@id"]),
             name=benchmark_metadata["name"],
             abstract=benchmark_metadata["abstract"],
             authors=tuple(str for author in benchmark_metadata["authors"]),
-            datasets=tuple(benchmark_datasets)
+            datasets=tuple(benchmark_datasets),
```

Comment:
You don't need datasets here. Instead have each BenchmarkDataset point back to its parent benchmark_uri.